improver/blending/spatial_weights.py

```diff
@@ -307,21 +303,28 @@ def process(self, cube_to_collapse, one_dimensional_weights_cube, blend_coord):
                 name given blend_coord. This cube contains 1D weights
                 that will be applied along the blend_coord but need
                 adjusting spatially based on missing data.
-            blend_coord (str):
-                A string containing the name of the coordinate that the
-                cube_to_collapse will be collapsed along. Also matches the
-                coordinate in one_dimensional_weights_cube.
 
         Returns:
             iris.cube.Cube:
-                A cube containing non-normalised spatial weights based on the
-                cube_to_collapsemask and the one_dimensional weights supplied.
-                Contains the dimensions, blend_coord, y, x.
+                A cube containing normalised spatial weights based on the
+                cube_to_collapse mask and the one_dimensional weights supplied.
+                Has dimensions: self.blend_coord, y, x.
         """
-        template_cube = self.create_template_slice(cube_to_collapse, blend_coord)
-        weights_from_mask = self.create_initial_weights_from_mask(template_cube)
-        weights_from_mask = self.smooth_initial_weights(weights_from_mask)
-        final_weights = self.multiply_weights(
-            weights_from_mask, one_dimensional_weights_cube, blend_coord
-        )
-        return final_weights
+        template_cube = self._create_template_slice(cube_to_collapse)
+        (self.blend_axis,) = template_cube.coord_dims(self.blend_coord)
+
+        weights = self._broadcast_weights(template_cube, one_dimensional_weights_cube)
```

Comment:
I prefer my implementation, if I'm honest.  It's cleaner, only does what it needs to, and minimises interdependencies between plugins.

Reply:
I mean, I could make it a utility and import it in both places if you like?

Reply:
> I prefer my implementation, if I'm honest. It's cleaner, only does what it needs to, and minimises interdependencies between plugins.

Except it destroys the array view on 1D data (memory increase) and goes via intermediate cubes + merge (more memory waste). Feel free to factor out `WeightedBlendAcrossWholeDimension.shape_weights` into a separate utility if it helps, but keep the implementation.

Reply:
OK, will factor out.

I assume the memory cleverness is down to the `iris.util.broadcast_to` call?  This I could easily put in the `_broadcast_weights` method.  But the `shape_weights` method should be at least two different methods at first glance - one to sort out the dimensions overall, the other to do the actual broadcasting - and I really don't want to be calling a ton of stuff I don't need (this causes problems down the line).

Reply:
Definitely one to factor out :-) then any optimisations applied to one will automatically be applied to the other.

With regard to the memory increase, anytime you alter a broadcasted array you save a new array in memory, instead of a (basically free) view of the array, so changing the dtype, adding a mask etc... Because of the way the weights need to be adjusted and masked at different x,y positions for each time used, I couldn't easily see a way to do this without creating an intermediate cube.